crates/torii/core/src/engine.rs

```diff
@@ -337,6 +256,11 @@ impl<P: Provider + Sync> Engine<P> {
                 self.process_block(block_number, blocks[&block_number]).await?;
                 last_block = block_number;
             }
+
+            // If the transaction has a world event, we update the cursor
+            if has_world_event {
+                pending_block_tx_cursor = Some(transaction_hash);
+            }
```

Comment:
since transactions only contain tx from `get_events` i think we don't need this check

Reply:
ah yes we do not anymore